tempesta_fw/http_msg.c

```diff
@@ -64,6 +90,58 @@ __tfw_http_msg_find_hdr(const TfwStr *hdr, const TfwStr array[], size_t size)
 }
 EXPORT_SYMBOL(__tfw_http_msg_find_hdr);
 
+/**
+ * Get header id in header table for header @hdr.
+ *
+ * TODO: return error if header is not allowed in @msg_type.
```

Comment:
Good topic to discuss and think about. Really, having `Server` header in a request doesn't look normal. However, there are Web applications setting custom headers. Moreover, there are proprietary soft (e.g. AntiBViruses) actively working with their clouds using HTTP and the soft sometimes sends quite strange traffic.

Maybe this is a subject for machine learning (to learn the normal set of HTTP headers). Maybe we should provide a configuration option to allow a user to specify which headers are allowed, like we're going to do this for HTTP strings alphabets in #628.

Reply:
This function checks only headers added by end user. The `(resp|req)_hdr_set` gives user opportunity to make very strange requests and responses and even broke network interactions between agents and server. But user must know what he does. 

>    Maybe we should provide a configuration option to allow a user to specify which headers are allowed

It's already here: user can block specific headers in network by with `(resp|req)_hdr_set <SomeHeader>`; all the headers <SomeHeader> will be removed from forwarded traffic.